src/main/java/mods/railcraft/common/blocks/wayobjects/RoutingLogic.java

```diff
@@ -220,8 +238,8 @@ public AND(Condition a, Condition b) {
         }
 
         @Override
-        public boolean matches(IRoutingTile tile, EntityMinecart cart) {
-            return a.matches(tile, cart) && b.matches(tile, cart);
+        public int evaluate(IRoutingTile tile, EntityMinecart cart) {
+            return (a.evaluate(tile, cart) != 0 ? b.evaluate(tile, cart) : valueForBool(false));
```

Comment:
Ooh, bitwise could be fun. That'd mean OR (`|`) could let you check a couple independent conditions at once (anything we have a leaf condition for), as long as you could come up with a way to decode them later (e.g. limited ability with analog controller box). The only issue is that it doesn't mesh well with redstone wire signal drop, though you could always set the low bit(s) to 1s to give a little bit of distance to work with.